internal/server/data/access_key.go

```diff
@@ -297,19 +295,47 @@ func ValidateRequestAccessKey(tx *Transaction, authnKey string) (*models.AccessK
 		if now.After(t.InactivityTimeout) {
 			return nil, ErrAccessInactivityTimeout
 		}
-
-		origTimeout := t.InactivityTimeout
 		t.InactivityTimeout = now.Add(t.InactivityExtension)
-		// Throttle updates when the key is used frequently. Uses the
-		// same value as server.lastSeenUpdateThreshold.
-		if t.InactivityTimeout.Sub(origTimeout) > 2*time.Second {
-			if err := UpdateAccessKey(tx, t); err != nil {
-				return nil, err
-			}
-		}
 	}
 
-	return t, nil
+	err = updateAccessKeyLastSeenAt(tx, t)
```

Comment:
This also fixes the problem where only access keys with an `InactivityTimeout` have the correct `LastSeenAt`. With this change we update the access key even if it doesn't have an `InactivityTimeout`, so the artificial `LastSeenAt` (based on `UpdatedAt`) should be correct now.

Reply:
Great fix!!